crates/sui-core/src/authority/authority_store_pruner.rs

```diff
@@ -326,4 +340,105 @@ mod tests {
         ma::assert_le!(after_compaction_size, before_compaction_size);
         Ok(())
     }
+
+    #[tokio::test]
+    async fn ensure_tombstone_fragmentation_in_stack_frame() -> Result<(), anyhow::Error> {
+        // This test writes a bunch of objects to objects table, invokes pruning on it and
+        // then does a bunch of get(). We open the db with `ignore_range_delete` set to false.
+        // We then record a cpu profile of the `get()` calls and find a range fragmentation stack frame
+        // in it.
```

Comment:
the two new test cases share a lot of duplicate code, can we refactor them?

Reply:
sure, fixed it